frontend/src/lib/components/BranchButton.svelte

```diff
@@ -0,0 +1,384 @@
+	 *
+	 * @throws Will throw an error if the response from the API is not successful. The error object
+	 * will include the status code and any additional error message from the API.
+	 *
+	 * @example
+	 * async function main() {
+	 *   try {
+	 *     const branches = await fetchExistingBranches();
+	 *     console.log('Existing branches:', branches);
+	 *   } catch (err) {
+	 *     console.error('Failed to fetch branches:', err);
+	 *   }
+	 * }
+	 */
+	async function fetchExistingBranches() {
+		const response = await fetch(`${apiAddress}/api/branches`, {
+			method: 'GET',
+			credentials: 'include',
+			headers: {
+				'Content-Type': 'application/json'
+			}
+		});
+
+		// Check if the response is successful
+		if (!response.ok) {
+			// Handle error without try-catch
+			response.json().then((errorMessage) => {
+				console.error('Failed to fetch branches:', errorMessage);
+				addToast({
+					message: `Error fetching branches: ${response.statusText}. ${JSON.stringify(errorMessage)}`,
+					type: ToastType.Error,
+					dismissible: true
+				});
+			});
+			return; // Exit if response is not OK
+		}
+
+		// Extract and set the branches if the response is successful
+		response.json().then((data) => {
+			if (data.data && data.data.branches) {
+				// Map through branches to filter out protected ones and extract branch names
+				nonProtectedBranches = data.data.branches
+					.filter((branch: string) => !branch.includes('(protected)')) // Filter out protected branches
+					.map((branch: string) => branch.split(' (')[0]); // Extract just branch names
+				protectedBranches = data.data.branches
+					.filter((branch: string) => branch.includes('(protected)')) // Filter out protected branches
+					.map((branch: string) => branch.split(' (')[0]); // Extract just branch names
+			} else {
+				protectedBranches = []; // Reset if no branches found
+				nonProtectedBranches = [];
+			}
+		});
+	}
+
+	onMount(() => {
+		fetchExistingBranches();
+	});
+
+	async function setBranchName(input: string) {
+		// Define validation rules
+		const maxLength = 255; // Maximum length for branch name
+		const invalidCharacters = /[~^:?*<>|]/; // Invalid special characters
+		const startsWithLetterOrNumber = /^[a-zA-Z0-9]/; // Starts with letter or number
+		const containsSpaces = /\s/; // Contains spaces
+
+		// Define an array of rules
+		const rules = [
+			'Branch names must start with a letter or a number.',
+			'Branch names cannot contain spaces. Use dashes (-) or underscores (_) instead.',
+			'Branch names cannot contain special characters like ~, ^, :, ?, *, and others.',
+			`Branch names must be shorter than ${maxLength} characters.`
+		];
+
+		const isValidBranchName = (name: string) => {
+			return (
+				!containsSpaces.test(name) &&
+				startsWithLetterOrNumber.test(name) &&
+				!invalidCharacters.test(name) &&
+				name.length <= maxLength
+			);
+		};
+
+		if (!input) return;
```

Comment:
This should be at the very top of the function, you don't need to start validation at all if no input is given